// @mantine/rte/cjs/components/Toolbar/controls.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var icons = require('@tabler/icons');

const CONTROLS = {
  code: {
    icon: icons.IconCode,
    controls: "code"
  },
  codeBlock: {
    icon: icons.IconCode,
    controls: "code-block"
  },
  bold: {
    icon: icons.IconBold,
    controls: "bold"
  },
  italic: {
    icon: icons.IconItalic,
    controls: "italic"
  },
  underline: {
    icon: icons.IconUnderline,
    controls: "underline"
  },
  strike: {
    icon: icons.IconStrikethrough,
    controls: "strike"
  },
  unorderedList: {
    icon: icons.IconList,
    controls: "list",
    value: "bullet"
  },
  orderedList: {
    icon: icons.IconListNumbers,
    controls: "list",
    value: "ordered"
  },
  link: {
    icon: icons.IconLink,
    controls: "link"
  },
  image: {
    icon: icons.IconPhoto,
    controls: "image"
  },
  clean: {
    icon: icons.IconClearFormatting,
    controls: "clean"
  },
  alignCenter: {
    icon: icons.IconAlignCenter,
    controls: "align",
    value: "center",
    noActive: true
  },
  alignLeft: {
    icon: icons.IconAlignLeft,
    controls: "align",
    value: "",
    noActive: true
  },
  alignRight: {
    icon: icons.IconAlignRight,
    controls: "align",
    value: "right",
    noActive: true
  },
  video: {
    icon: icons.IconBrandYoutube,
    controls: "video"
  },
  h1: {
    icon: icons.IconH1,
    controls: "header",
    value: "1"
  },
  h2: {
    icon: icons.IconH2,
    controls: "header",
    value: "2"
  },
  h3: {
    icon: icons.IconH3,
    controls: "header",
    value: "3"
  },
  h4: {
    icon: icons.IconH4,
    controls: "header",
    value: "4"
  },
  h5: {
    icon: icons.IconH5,
    controls: "header",
    value: "5"
  },
  h6: {
    icon: icons.IconH6,
    controls: "header",
    value: "6"
  },
  sup: {
    icon: icons.IconSuperscript,
    controls: "script",
    value: "super"
  },
  sub: {
    icon: icons.IconSubscript,
    controls: "script",
    value: "sub"
  },
  blockquote: {
    icon: icons.IconBlockquote,
    controls: "blockquote"
  }
};

exports.CONTROLS = CONTROLS;
